.gitignore

```diff
@@ -1,19 +1,44 @@
+.coverage
 *.egg-info
+*.log
 *.mo
-*.pyc
-*~
-.installed.cfg
-.mr.developer.cfg
+*.py?
+*.swp
+# dirs
 bin/
-coverage/
+buildout-cache/
 develop-eggs/
-downloads/
 eggs/
-parts/
-var/
+htmlcov/
 include/
 lib/
-Makefile
-.pydevproject
-.project
-.settings
+local/
+node_modules/
+parts/
+src/*
+dist/*
+test.plone_addon/
+var/
+# files
+.installed.cfg
+.mr.developer.cfg
+lib64
+log.html
+output.xml
+pip-selfcheck.json
+report.html
+.vscode/
+.tox/
+reports/
+/share/
+/local
+*project
+/.settings/
+# excludes
+!.coveragerc
```

Comment:
Are the `!.*` lines needed?

Reply:
They are necessary to keep these files in version control. For example, the `.coveragerc` file must be versioned. I just used the `bobtemplates.plone` configuration:

https://github.com/plone/bobtemplates.plone/blob/05026f80839ecb84506a729082a63df093fff72c/bobtemplates/plone/addon/.gitignore.bob#L34

Reply:
Not really unless you have some global gitignore weirdness
```
[ale@emily collective.recaptcha]$ touch .coveragerc 
[ale@emily collective.recaptcha]$ git status
On branch master
Your branch is up to date with 'origin/master'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .coveragerc

nothing added to commit but untracked files present (use "git add" to track)
```

I am not a big fun of this change but anyway...

In addition I found that those exclude rules came after this change that proposed to ignore `.*` https://github.com/plone/bobtemplates.plone/commit/15b6065c31f21d1b1ede9c3638f98d00ed39afe4